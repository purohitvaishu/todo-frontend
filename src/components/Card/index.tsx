/* eslint-disable no-unused-vars */
/* eslint-disable @typescript-eslint/no-unused-vars */
import React, { FunctionComponent } from "react";
import { makeStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import Typography from "@material-ui/core/Typography";
import { Divider, CardHeader } from "@material-ui/core";

const useStyles = makeStyles({
  root: {
    paddingLeft: 26,
    color: "white",
    cursor: "pointer",
    minHeight: 260,
    paddingRight: 4,
    background: "transparent",
    boxShadow: "none"
  },
  title: {
    fontSize: 14,
    paddingBottom: 10,
    textAlign: "end",
    color: "white"
  },
  cardHeader: {
    paddingLeft: 0
  }
});

export interface Bucket {
  title: string;
  description: string;
  completed: boolean;
}

type Props = {
  data: {
    bucketname: string;
    data: Bucket[];
  };
};

const CardComponent: FunctionComponent<Props> = (props: Props) => {
  const classes = useStyles();
  const { data } = props;

  return (
    <Card className={classes.root}>
      <CardHeader title={data.bucketname} className={classes.cardHeader} />
      <Divider />
      <CardContent className="scroll">
        {data &&
          data.data &&
          data.data.map(values => (
            <Typography
              className={classes.title}
              color="textSecondary"
              gutterBottom
              style={
                values.completed
                  ? { textDecoration: "line-through" }
                  : { textDecoration: "none" }
              }
            >
              {values.title}
            </Typography>
          ))}
      </CardContent>
    </Card>
  );
};

export default CardComponent;
