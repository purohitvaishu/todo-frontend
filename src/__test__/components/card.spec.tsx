/* eslint-disable react/jsx-props-no-spreading */
import React from "react";
import { shallow } from "enzyme";
import Card from "../../components/Card";

describe("Test: Bucket", () => {
  test("container", () => {
    const props = {
      data: {
        bucketname: "",
        data: []
      }
    };

    const result = shallow(<Card {...props} />);
    expect(result.debug()).toMatchSnapshot();
  });
});
