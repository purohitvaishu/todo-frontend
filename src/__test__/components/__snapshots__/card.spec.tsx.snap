// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test: Bucket container 1`] = `
"<WithStyles(ForwardRef(Card)) className=\\"makeStyles-root-1\\">
  <WithStyles(ForwardRef(CardHeader)) title=\\"\\" className=\\"makeStyles-cardHeader-3\\" />
  <WithStyles(ForwardRef(Divider)) />
  <WithStyles(ForwardRef(CardContent)) className=\\"scroll\\" />
</WithStyles(ForwardRef(Card))>"
`;
